lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -0,0 +1,49 @@
+"""
+This file contains implementation override of SearchFilterGenerator which will allow
+    * Filter by all courses in which the user is enrolled in
+"""
+from datetime import datetime
+
+from student.models import CourseEnrollment
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+
+from search.filter_generator import SearchFilterGenerator
+from search.utils import DateRange
+from openedx.core.djangoapps.course_groups.cohorts import get_cohort
+from openedx.core.djangoapps.course_groups.models import CourseUserGroupPartitionGroup
+
+
+class LmsSearchFilterGenerator(SearchFilterGenerator):
+    """ SearchFilterGenerator for LMS Search """
+
+    def filter_dictionary(self, **kwargs):
+        """ base implementation which filters via start_date """
+        filter_dictionary = {"start_date": DateRange(None, datetime.utcnow())}
```

Comment:
I think it would be better to call the base class implementation here to get the start date involved - that way, if it ever changes, then the changes will propagate